### ISSUE TITLE
1.3.1 Stealth enhancement

### ISSUE THREAD

**Issue by L0ck0n** [PapaJoesSoup/BDArmory#365]
Stealth feature is an awesome addition to the game. But it is quite simple, because it must be resource-efficient (optimization). Current RCS is based (calculated) on three pictures (frontal, lateral, ventral).  Now the plane, which has low frontal RCS, still has large overall RCS (and is easy to detect), because it has a large ventral or lateral RCS. Real-time calculating, making these grayscale images in the game would be too demanding. But I have an idea: the game could calculate in real time only the angle (in three dimensions) at which "plane A" is flying relative to the "plane B" radar. An angle in the X, Y, Z axis could be set to determine whether the "plane A" is facing forward or sideways to the "plane B" radar. If "plane A" would fly to the "plane B" frontally, the game would select premade (in hangar) RCS value based on frontal grayscale of "plane A". "Plane B" radar would take into account only the frontal RCS of the "plane A". The game would have to calculate in real time only the angle between planes or objects. Such system shouldn't be resource-hungry. Here is a picture that shows how it could work:
![new rcs system conception](https://user-images.githubusercontent.com/30219674/32806242-24082588-c98c-11e7-97e6-88a66c761158.png)

**Comment by Deathpuff12** [comment]
Isn't this how it works right now? If not, they're really

**Comment by L0ck0n** [comment]
I think that its now based on precalculated total average (frontal+ventral+lateral=average). It could be based on direction (vessel is exposing side to the radar= only precalculated side RCS is taken into account). Now completely frontal-stealthy plane is easily detectable even if its showing only stealthy front to the radar, because it has large ventral RCS.
![b-2](https://user-images.githubusercontent.com/30219674/32842790-4baf60c8-ca1e-11e7-85bf-b8dc18942361.png)
This plane has low frontal RCS and is easy to detect because it has high ventral RCS value.

**Comment by gomker** [comment]
I will have to leave it to @TheDogKSP  as the authority, are we taking the average of these views?
Also, in reality, would the "ventral" view always be that wide, that is a very wide surface, and how often would you be right underneath a radar? For the most part it would be your lateral / frontal view.

**Comment by gomker** [comment]
@L0ck0n  Also very nice craft, you got a download for that :)

**Comment by TheDogKSP** [comment]
Yes, current logic takes the avg as one static number which is used for comparison against a radars capability curves.

**Comment by L0ck0n** [comment]
You don't need to be under the radar to show the "ventral" RCS value to it, you can fly up or down and the top or bottom of the plane will be exposed.
![rcs](https://user-images.githubusercontent.com/30219674/32912013-1e140114-cb0e-11e7-9132-14d4357bcfb7.png)
P.S. thx, here it is:
https://kerbalx.com/Lock0n/craft

**Comment by Paulo-D2000** [comment]
Nice! Maybe try Triplanar Projection ? Its like projecting the maps as one cube to the vessel itself...
